src/extensions/stores/cluster-store.ts

```diff
@@ -0,0 +1,121 @@
+import { clusterStore as internalClusterStore, ClusterId } from "../../common/cluster-store";
+import type { ClusterModel } from "../../common/cluster-store";
+import { Cluster } from "../../main/cluster";
+import { Singleton } from "../core-api/utils";
+
+export { Cluster } from "../../main/cluster";
+export type { ClusterModel, ClusterId } from "../../common/cluster-store";
+
+/**
+ * Store for all added clusters
+ */
+export class ClusterStore extends Singleton {
+
+  /**
+   * Active cluster id
+   */
+  activeClusterId = internalClusterStore.activeCluster;
+
+  /**
+   * Map of all clusters
+   */
+  clusters = internalClusterStore.clusters;
```

Comment:
I am pretty sure that this would definitely break if either side were to for instance do a `= []` instead of `.length = 0`.

Reply:
Right, maybe needs to be `get`?